File: ComplementaryScripts/loadKmxModel.py
# Benjamin J. Sanchez
# Simonas Marcisauskas, 2019-11-07 - adaptation for Kluyveromyces_marxianus-GEM

# -*- coding: utf-8 -*-
import cobra

# Load model
model = cobra.io.read_sbml_model('../ModelFiles/xml/kmxGEM.xml')

# Correct metabolite ids:
for met in model.metabolites:
    met.id = met.id.replace('__91__', '_')
    met.id = met.id.replace('__93__', '')

File: ComplementaryScripts/loadKmxModel.m
function model = loadKmxModel
% loadKmxModel
%	Loads model from hard-coded location and adds the rxnGeneMat to the
%	structure
%
%	Usage: loadKmxModel
%
%	Based on function loadYeastModel written by Benjamin Sanchez
%	(https://github.com/SysBioChalmers/yeast-GEM)
%	Simonas Marcisauskas, 2019-11-10 - adaptation for
%	Kluyveromyces_marxianus-GEM
%

scriptFolder = fileparts(which(mfilename));
currentDir = cd(scriptFolder);
model = readCbModel('../ModelFiles/xml/kmxGEM.xml');
model = buildRxnGeneMat(model);
cd(currentDir)

end
